client/my-sites/checkout/composite-checkout/components/checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { isPlan } from '@automattic/calypso-products';
+import { FormStatus, useFormStatus } from '@automattic/composite-checkout';
+import formatCurrency from '@automattic/format-currency';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import debugFactory from 'debug';
+import { useSelector } from 'react-redux';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import PromoCardCTA from 'calypso/components/promo-section/promo-card/cta';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { useGetProductVariants } from '../../hooks/product-variants';
+import {
+	getItemVariantCompareToPrice,
+	getItemVariantDiscountPercentage,
+} from '../item-variation-picker/variant-price';
+
+import './style.scss';
+
+const debug = debugFactory( 'calypso:checkout-sidebar-plan-upsell' );
+
+export function CheckoutSidebarPlanUpsell() {
+	const { formStatus } = useFormStatus();
+	const isFormLoading = FormStatus.READY !== formStatus;
+	const { __ } = useI18n();
+	const cartKey = useCartKey();
+	const { responseCart, replaceProductInCart } = useShoppingCart( cartKey );
+	const siteId = useSelector( getSelectedSiteId );
+	const plan = responseCart.products.find( ( product ) => isPlan( product ) );
+	const variants = useGetProductVariants( siteId ?? undefined, plan?.product_slug ?? '' );
+
+	if ( ! plan ) {
+		debug( 'no plan found in cart' );
+		return null;
+	}
+
+	const biennialVariant = variants?.find( ( product ) => product.termIntervalInMonths === 24 );
+	const currentVariant = variants?.find( ( product ) => product.productId === plan.product_id );
+
+	if ( ! biennialVariant ) {
+		debug( 'plan in cart has no biennial variant; variants are', variants );
+		return null;
+	}
+
+	if ( ! currentVariant ) {
+		debug( 'plan in cart has no current variant; variants are', variants );
+		return null;
+	}
+
+	if ( biennialVariant.productId === plan.product_id ) {
+		debug( 'plan in cart is already biennial' );
+		return null;
+	}
+
+	const onUpgradeClick = () => {
```

Comment:
Could we add a Tracks event to this (or possibly a default one to `PromoCardCTA` with a prop that identifies which nudge it was)?